(SYS)NEVM updates

walkjivefly opened the pull request:
exrproxy changes required for https://github.com/blocknetdx/eth-payment-processor/pull/29

ConanMishler reviewed (changes requested):
So I built `exrproxy` locally from the `feature-SYS` branch, but the logs for the `exrproxy-env-xr_proxy-1` container keep printing this error:
```
[error] 10#10: *663 "/etc/nginx/html/index.html" is not found (2: No such file or directory)
```
Do you see that error also when you do:
```
docker logs exrproxy-env-xr_proxy-1 -n 10
```
???

I don't know if it's that error or something else, but when I `request_project` from the VPS, I'm still seeing:
```
    "payment_amount_tier1_sysblock": null,
    "payment_amount_tier2_sysblock": null,
```
and I'm still not seeing:
```
    "payment_nevm_address": "0x..."

walkjivefly commented:
> So I built `exrproxy` locally from the `feature-SYS` branch, but the logs for the `exrproxy-env-xr_proxy-1` container printed this error:
> 
> ```
> ERROR - Failed to load evm_passthrough plugin: column payment.sys_address does not exist
> LINE 1: ...payment"."eth_address", "payment"."avax_address", "payment"....
> ```
> 
> Then it keeps printing this error:
> 
> ```
> [error] 10#10: *663 "/etc/nginx/html/index.html" is not found (2: No such file or directory)
> ```
> 
> Do you see that error also when you do:
> 
> ```
> docker logs exrproxy-env-xr_proxy-1 -n 10
> ```
> 
> ???
> 
> I don't know if it's that error or something else, but when I `request_project` from the VPS, I'm still seeing:
> 
> ```
>     "payment_amount_tier1_sysblock": null,
>     "payment_amount_tier2_sysblock": null,
> ```
> 
> and I'm still not seeing:
> 
> ```
>     "payment_nevm_address": "0x..."
> ```
> 
> Note, I tried restarting the `exrproxy-env-payment_db-1`, `exrproxy-env-payment-1` and `exrproxy-env-xr_proxy-1` containers but nothing helped.

I've got 2 VPSs running the SYS NEVM stuff. On one I wasn't seeing these errors. I found this morning I do see them on the second instance. I couldn't work out why, but I've been updating things piecemeal on this machine to avoid restarting the NEVM event processor. Well I bit the bullet, 
shut down all the containers, 
removed the old images, 
re-ran `builder.py`, 
tweaked the docker-compose  and brought it all back up. Errors gone.

But then I realised I'd forgotten to pick up all my latest images. So shut it all down again, updated the docker-compose.yml and brought it back up and yes, I'm getting the same errors now. Obviously I've gotten something out of sync between tinkering in 4 separate repos. I can't even find the string "sys_address" in any of them but I'm guessing it's something to do with the somewhat inconsistent use of "SYS" and "NEVM" in field/variable names. I'll shout when I get this instance working right again...

ConanMishler commented on plugins/evm_passthrough/routes.py:
```diff
@@ -56,7 +56,8 @@ def handle_request(evm, project_id, path=None):
         'API-TOKENS-USED': str(g.project.used_api_tokens),
-        'API-TOKENS-REMAINING': str(g.project.api_token_count - g.project.used_api_tokens)
+        'API-TOKENS-REMAINING': str(g.project.api_token_count - g.project.used_api_tokens),
+        'EXPIRY-DATE':str(g.project.expires)
```
Nice idea to return the expiration date in the header. I was actually planning to add the same feature. One small request, however:  Can we change this to be like this:
```python
'EXPIRATION-DATE':str(g.project.expires)
```
The reason is because there is already an "expiry_time" field returned by the `request_project` method, which represents the time when the prices quoted by the `request_project` method expire. To avoid confusion between the two kinds of expiry/expiration, I'm trying to keep their names different (expiry vs expiration) in the API docs.

walkjivefly commented:
It should be merged together with

- https://github.com/blocknetdx/XQuery/pull/49
- https://github.com/blocknetdx/eth-payment-processor/pull/29
- https://github.com/blocknetdx/exrproxy-env/pull/187

walkjivefly reviewed (commented):
Thanks for finding this. I was not looking forward to tracking it down when it wasn't an obvious code error.

It builds OK and I'm not getting the 502 errors now.

walkjivefly reviewed (commented):
LGTM

ConanMishler reviewed (approved):
Seems to work based on my testing, and doesn't break anything.
